dotnet/notebooks/06-vector-stores-and-embeddings.ipynb

```diff
@@ -0,0 +1,464 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "dotnet_interactive": {
+     "language": "csharp"
+    },
+    "polyglot_notebook": {
+     "kernelName": "csharp"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "var glossaryEntries = new List<Glossary>()\n",
+    "{\n",
+    "    new Glossary() \n",
+    "    {\n",
+    "        Key = 1,\n",
+    "        Term = \"API\",\n",
+    "        Definition = \"Application Programming Interface. A set of rules and specifications that allow software components to communicate and exchange data.\"\n",
+    "    },\n",
+    "    new Glossary() \n",
+    "    {\n",
+    "        Key = 2,\n",
+    "        Term = \"Connectors\",\n",
+    "        Definition = \"Connectors allow you to integrate with various services provide AI capabilities, including LLM, AudioToText, TextToAudio, Embedding generation, etc.\"\n",
+    "    },\n",
+    "    new Glossary() \n",
+    "    {\n",
+    "        Key = 3,\n",
+    "        Term = \"RAG\",\n",
+    "        Definition = \"Retrieval Augmented Generation - a term that refers to the process of retrieving additional data to provide as context to an LLM to use when generating a response (completion) to a user's question (prompt).\"\n",
+    "    }\n",
+    "};"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we want to perform a vector search on our records in a database, initializing just a key and data properties is not enough, we also need to generate and initialize vector properties. For that, we can use `ITextEmbeddingGenerationService` which we already registered above:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "dotnet_interactive": {
+     "language": "csharp"
+    },
+    "polyglot_notebook": {
+     "kernelName": "csharp"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "using Microsoft.SemanticKernel.Embeddings;\n",
+    "\n",
+    "#pragma warning disable SKEXP0001\n",
+    "\n",
+    "var textEmbeddingGenerationService = kernel.GetRequiredService<ITextEmbeddingGenerationService>();\n",
+    "\n",
+    "var tasks = glossaryEntries.Select(entry => Task.Run(async () =>\n",
+    "{\n",
+    "    entry.DefinitionEmbedding = await textEmbeddingGenerationService.GenerateEmbeddingAsync(entry.Definition);\n",
+    "}));\n",
+    "\n",
+    "await Task.WhenAll(tasks);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Upsert records\n",
+    "\n",
+    "Now our glossary records are ready to be inserted into database. For that, we can use `collection.UpsertAsync` or `collection.UpsertBatchAsync` methods. Note that this operation is idempotent - if record with specific key doesn't exist, it will be inserted. If it already exists, it will be updated. As a result, we should receive a keys of recently upserted records:"
```

Comment:
```suggestion
    "Now our glossary records are ready to be inserted into the database. For that, we can use `collection.UpsertAsync` or `collection.UpsertBatchAsync` methods. Note that this operation is idempotent - if a record with a specific key doesn't exist, it will be inserted. If it already exists, it will be updated. As a result, we should receive the keys of the upserted records:"
```